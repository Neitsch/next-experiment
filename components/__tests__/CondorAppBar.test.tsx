import { shallow } from "enzyme";
import React from "react";

describe("AppBar", () => {
  beforeEach(() => {
    jest.resetModules();
  });
  describe("unauthenticated", () => {
    let AppBarRendered = null;
    beforeEach(() => {
      const CondorAppBar = require("../CondorAppBar").CondorAppBar;
      const AppBarComponent = (
        <CondorAppBar isAuthenticated={false} classes={{ flex: "flex" }} />
      );
      AppBarRendered = shallow(AppBarComponent);
    });
    it("renders", () => {
      expect(AppBarRendered).toMatchSnapshot();
    });
  });
  describe("authenticated", () => {
    let AppBarRendered = null;
    beforeEach(() => {
      const CondorAppBar = require("../CondorAppBar").CondorAppBar;
      const AppBarComponent = (
        <CondorAppBar isAuthenticated={true} classes={{ flex: "flex" }} />
      );
      AppBarRendered = shallow(AppBarComponent);
    });
    it("renders", () => {
      expect(AppBarRendered).toMatchSnapshot();
    });
    it("Open Menu", () => {
      const method = AppBarRendered.find("#accountButton").props().onClick;
      const anchorElem = jest.fn();
      method({ currentTarget: anchorElem });
      expect(AppBarRendered.state("openAccountMenu")).toEqual(anchorElem);
    });
  });
});
