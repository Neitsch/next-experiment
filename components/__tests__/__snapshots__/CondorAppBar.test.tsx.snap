// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar authenticated renders 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
    >
      <pure(Menu) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      className="CondorAppBar-flex-1"
      color="inherit"
      variant="title"
    >
      Condor Club
    </WithStyles(Typography)>
    <div>
      <WithStyles(IconButton)
        aria-haspopup="true"
        aria-owns="menu-appbar"
        color="inherit"
        id="accountButton"
        onClick={[Function]}
      >
        <pure(AccountCircle) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`AppBar unauthenticated renders 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
    >
      <pure(Menu) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      className="CondorAppBar-flex-1"
      color="inherit"
      variant="title"
    >
      Condor Club
    </WithStyles(Typography)>
    <WithStyles(Button)
      color="inherit"
      id="login"
    >
      Login
    </WithStyles(Button)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
